### DISCUSSION TITLE
New data sources for OpenLineage in Apache Airflow

### DISCUSSION THREAD

**Discussion by pletier35** [OpenLineage/OpenLineage#483] [Q&A]
My org uses Apache Airflow to orchestrate hundreds of jobs that feed our Bigquery data warehouse. We utilize a handful of ETL patterns in GCP. We are exploring OpenLineage as our primary lineage consolidation tool.

However, we have found that OpenLineage does not capture lineage for many of our ETL jobs (e.g. DataflowOperator, GoogleCloudStorageToBigQueryOperator, and BigQueryToBigQueryOperator).

We would like to know if there are any plans to integrate new Apache Airflow Data Sources such as Google Cloud Storage, Dataflow, etc? If so, is there a release timeline?

**Comment by mobuchowski** [comment]
Hey, why not contribute them? 
You can take a look at how [BigQueryExtractor](https://github.com/OpenLineage/OpenLineage/blob/main/integration/airflow/openlineage/airflow/extractors/bigquery_extractor.py) is implemented and base another bigquery operators on this.

If you can't contribute, you can always attach them to your instance using [custom extractors](https://github.com/OpenLineage/OpenLineage/tree/main/integration/airflow#custom-extractors).
